solutions/devsprint-leonardo-santos-2/FinanceApp/Components/ActivityListView.swift

```diff
@@ -6,61 +6,106 @@
 //
 
 import UIKit
+import Cartography
 
 protocol ActivityListViewDelegate: AnyObject {
-
     func didSelectedActivity()
 }
 
-class ActivityListView: UIView {
+final class ActivityListView: UIView {
 
+    //MARK: - Properties
     weak var delegate: ActivityListViewDelegate?
 
-    static let cellSize = CGFloat(82)
-
-    private let cellIdentifier = "ActivityCellIdentifier"
-
-    lazy var tableView: UITableView = {
+    static let cellSize = Constant.cellHeight
 
+    //MARK: - UI Components
+    private(set) lazy var tableView: UITableView = {
         let tableView = UITableView(frame: .zero)
-        tableView.translatesAutoresizingMaskIntoConstraints = false
-        tableView.register(ActivityCellView.self, forCellReuseIdentifier: self.cellIdentifier)
-        tableView.dataSource = self
         tableView.delegate = self
+        tableView.dataSource = self
+        tableView.register(
+            ActivityCellView.self,
+            forCellReuseIdentifier: ActivityCellView.identifier
+        )
         return tableView
     }()
 
-    init() {
-        super.init(frame: .zero)
-
-        backgroundColor = .white
-        addSubviews()
-        configureConstraints()
-
-        tableView.reloadData()
+    //MARK: - Initializers
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        commomInit()
     }
 
     required init?(coder: NSCoder) {
-        fatalError("init(coder:) has not been implemented")
+        super.init(frame: .zero)
+        commomInit()
+    }
+    
+    //MARK: - Helpers
+    private func commomInit() {
+        configureViewCode()
+        tableView.reloadData()
+    }
+    
+    private func actionProviderHandler(_ menus: [UIMenuElement], at indexPath: IndexPath) -> UIMenu? {
+        guard let cell = isActivityCell(at: indexPath) else { return .init() }
+        
+        let like = contextMenuButtonHandler(cell)
+        
+        let buttonChildrens: [UIAction] = [
+            .default(
+                title: Constant.openButtonTitle, icon: .link
+            ) { self.openAction($0, at: indexPath) },
+            .default(
+                title: like.title, icon: like.icon
+            ) { self.likeAction($0, at: indexPath) }
+        ]
+        return .init(
+            title: Constant.contextMenuTitle, image: nil,
+            identifier: nil, options: .displayInline,
+            children: buttonChildrens
+        )
+    }
+    
+    private func openAction(_ action: UIAction, at indexPath: IndexPath) {
+        delegate?.didSelectedActivity()
+    }
```

Comment:
Podemos remover esses parametros não utilizados?

```suggestion
    private func openAction() {
        delegate?.didSelectedActivity()
    }
```